commit c4c774f850c370413270fd07670593d08d403162
Author: P33M <p33m@github.com>
Date:   Mon Apr 8 12:45:23 2019 +0100

    dwc_otg: only do_split when we actually need to do a split
    
    The previous test would fail if the root port was in fullspeed mode
    and there was a hub between the FS device and the root port. While
    the transfer worked, the schedule mangling performed for high-speed
    split transfers would break leading to an 8ms polling interval.

diff --git a/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c b/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c
index fe8e8f841f03..2823dc9af63f 100644
--- a/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c
+++ b/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c
@@ -167,8 +167,10 @@ void qh_init(dwc_otg_hcd_t * hcd, dwc_otg_qh_t * qh, dwc_otg_hcd_urb_t * urb)
 	char *speed, *type;
 	int dev_speed;
 	uint32_t hub_addr, hub_port;
+	hprt0_data_t hprt;
 
 	dwc_memset(qh, 0, sizeof(dwc_otg_qh_t));
+	hprt.d32 = DWC_READ_REG32(hcd->core_if->host_if->hprt0);
 
 	/* Initialize QH */
 	qh->ep_type = dwc_otg_hcd_get_pipe_type(&urb->pipe_info);
@@ -191,9 +193,8 @@ void qh_init(dwc_otg_hcd_t * hcd, dwc_otg_qh_t * qh, dwc_otg_hcd_urb_t * urb)
 
 	qh->nak_frame = 0xffff;
 
-	if (((dev_speed == USB_SPEED_LOW) ||
-	     (dev_speed == USB_SPEED_FULL)) &&
-	    (hub_addr != 0 && hub_addr != 1)) {
+	if (hprt.b.prtspd == DWC_HPRT0_PRTSPD_HIGH_SPEED &&
+			dev_speed != USB_SPEED_HIGH) {
 		DWC_DEBUGPL(DBG_HCD,
 			    "QH init: EP %d: TT found at hub addr %d, for port %d\n",
 			    dwc_otg_hcd_get_ep_num(&urb->pipe_info), hub_addr,
@@ -204,7 +205,6 @@ void qh_init(dwc_otg_hcd_t * hcd, dwc_otg_qh_t * qh, dwc_otg_hcd_urb_t * urb)
 
 	if (qh->ep_type == UE_INTERRUPT || qh->ep_type == UE_ISOCHRONOUS) {
 		/* Compute scheduling parameters once and save them. */
-		hprt0_data_t hprt;
 
 		/** @todo Account for split transfers in the bus time. */
 		int bytecount =
@@ -219,7 +219,6 @@ void qh_init(dwc_otg_hcd_t * hcd, dwc_otg_qh_t * qh, dwc_otg_hcd_urb_t * urb)
 						    SCHEDULE_SLOP);
 		qh->interval = urb->interval;
 
-		hprt.d32 = DWC_READ_REG32(hcd->core_if->host_if->hprt0);
 		if (hprt.b.prtspd == DWC_HPRT0_PRTSPD_HIGH_SPEED) {
 			if (dev_speed == USB_SPEED_LOW ||
 					dev_speed == USB_SPEED_FULL) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1422_c4c774f850c370413270fd07670593d08d403162.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1422_c4c774f850c370413270fd07670593d08d403162.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
